crates/api_crud/src/comment/update.rs

```diff
@@ -63,15 +65,32 @@ impl PerformCrud for EditComment {
       return Err(LemmyError::from_message("no_comment_edit_allowed"));
     }
 
-    // Do the update
-    let content_slurs_removed =
-      remove_slurs(&data.content.to_owned(), &context.settings().slur_regex());
-    let comment_id = data.comment_id;
-    let updated_comment = blocking(context.pool(), move |conn| {
-      Comment::update_content(conn, comment_id, &content_slurs_removed)
-    })
-    .await?
-    .map_err(|e| LemmyError::from_error_message(e, "couldnt_update_comment"))?;
+    if let Some(distinguished) = data.distinguished {
+      // Verify that only a mod or admin can distinguish a comment
+      is_mod_or_admin(
+        context.pool(),
+        local_user_view.person.id,
+        orig_comment.community.id,
+      )
+      .await?;
```

Comment:
So this means that admins/mods can distuingish any comment by another user? Then there should be some indication who did it.

But actually, i think that would make it less of a mod tool, and more like giving gold on reddit. Imo it would be better if admins/mods could only distinguish their own comments (when they are giving a warning or similar).

Reply:
No, there's a check right above this one that makes sure they can only do these actions on their own comments.